html/apps/barchart/services/analyticService.js

```diff
@@ -0,0 +1,202 @@
+        function getNewTweets(statuses){
+            // initial oldest date from first element of last poll
+            var prevNewest = new Date($localStorage.tweets[0].created_at);
+            var idx =0;
+            var tweetDate= new Date(statuses[idx].created_at);
+            var currOldest = new Date(statuses[statuses.length-1].created_at);
+            
+            var newTweets =[];
+            // if the whole array is new prepend to localStorage tweets array
+            if(prevNewest < currOldest){
+                $localStorage.tweets = statuses.concat($localStorage.tweets);
+                newTweets = statuses;
+            } else {
+                // else prepend only new tweets to localStorage tweets array in desc order
+                while(tweetDate > prevNewest && idx !== statuses.length){
+                    newTweets.push(statuses[idx]);
+                    // // console.log($localStorage.tweets);
+                    idx++;
+                    tweetDate= new Date(statuses[idx].created_at);
+                }
+            }
+            return newTweets;
+        }
+        
+        function updateWordFreq(newTweets){
+            
+            // sWords is an array of selected words from tweet
+            var sWords = newTweets
+            .map(function(tweet){ 
+                // remove image tags, punctuation, links
+                return tweet.text.toLowerCase()
+                .replace(/<img\b[^>]*>.*?>?/g,'').replace(/[!,;.]/g,'')
+                .replace(/https?:\/\/[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|]/gi,''); 
+            })
+            .join(" ")
+            .split(/[\s\/]+/g);
+            // console.log(sWords);
+            
+            
+            // count w/ duplicates
+            var iWordsCount = sWords.length; 
+            
+            // object of english words to ignore
+            var ignore = ['and','the','to','a','of','for','as','i','with','it','is','on','that','this','can','in','be','has','if'];
+            ignore = (function(){
+                var o = {}; // object prop checking > in array checking
+                var iCount = ignore.length;
+                for (var i=0;i<iCount;i++){
+                    o[ignore[i]] = true;
+                }
+                return o;
+            }());
+            
+            // use an object for the math
+            var counts = {}; 
+            for (var i=0; i<iWordsCount; i++) {
+                var sWord = sWords[i];
+                if (!ignore[sWord]) {
+                    counts[sWord] = counts[sWord] || 0;
+                    counts[sWord]++;
+                }
+            }
+            // // console.log("counts: ", counts);
+            
+            if(Object.keys(counts).length === 0 && counts.constructor === Object){
+                $localStorage.wordFreq = counts;
+            } else {
+                for (sWord in counts){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["sWord" used outside of binding context.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2626095151/issues/source?bid=3288751&fileBranchId=3313532#l81)